CMakeLists.txt

```diff
@@ -28,10 +30,17 @@ if (NOT CMAKE_CXX_FLAGS)
         add_definitions("-DNOGDI=1")
         add_definitions("-DNOMINMAX=1")
     else()
-        set(CMAKE_CXX_FLAGS "-std=c++11 -Wall")
+        set(CMAKE_CXX_FLAGS "-Wall")
     endif()
 endif()
 
+if(USE_CPP17)
+    set(CMAKE_CXX_STANDARD 17)
+    add_definitions("-D_USE_CPP17")
```

Comment:
I don't think this is enough. There needs to be a record somewhere (e.g. in a generated header file) that stores this and code should include that file to figure out this flag.

If you don't do that, you can have the library being compiled with this flag on but an application that uses the library not knowing (for obvious reasons) that they need to set this macro manually, which will break badly because from one's perspective you're using std types and from the other's boost ones.